txnsync/msgorderingheap.go

```diff
@@ -98,3 +103,30 @@ func (p *messageOrderingHeap) pop() (msg incomingMessage, err error) {
 	entry := heap.Pop(p).(messageHeapItem)
 	return incomingMessage(entry), nil
 }
+
+func (p *messageOrderingHeap) compress() {
```

Comment:
maybe rename "compress" -> "compact" ? The "compress" keyword implies a reversible packing action, which isn't the case here.